daemon/swap_daemon_test.go

```diff
@@ -69,6 +78,69 @@ func getSwapFactoryAddress(t *testing.T, ec *ethclient.Client) ethcommon.Address
 	return swapFactoryAddr
 }
 
+func privKeyToAddr(privKey *ecdsa.PrivateKey) ethcommon.Address {
+	return crypto.PubkeyToAddress(*privKey.Public().(*ecdsa.PublicKey))
+}
+
+func transfer(t *testing.T, fromKey *ecdsa.PrivateKey, toAddress ethcommon.Address, ethAmount *apd.Decimal) {
+	ctx := context.Background()
+	ec, chainID := tests.NewEthClient(t)
+	fromAddress := privKeyToAddr(fromKey)
+
+	gasPrice, err := ec.SuggestGasPrice(ctx)
+	require.NoError(t, err)
+
+	nonce, err := ec.PendingNonceAt(ctx, fromAddress)
+	require.NoError(t, err)
+
+	weiAmount := coins.EtherToWei(ethAmount).BigInt()
+
+	tx := ethtypes.NewTx(&ethtypes.LegacyTx{
+		Nonce:    nonce,
+		To:       &toAddress,
+		Value:    weiAmount,
+		Gas:      TransferGas,
+		GasPrice: gasPrice,
+	})
+	signedTx, err := ethtypes.SignTx(tx, ethtypes.LatestSignerForChainID(chainID), fromKey)
+	require.NoError(t, err)
+
+	err = ec.SendTransaction(ctx, signedTx)
+	require.NoError(t, err)
+	_, err = block.WaitForReceipt(ctx, ec, signedTx.Hash())
+	require.NoError(t, err)
+}
+
+// minimumFundAlice gives Alice enough ETH to do everything but relay a claim
+func minimumFundAlice(t *testing.T, ec extethclient.EthClient, providesAmt *apd.Decimal) {
+	fundingKey := tests.GetTakerTestKey(t)
+
+	// When this comment was written, sample gas costs were:
+	//   newSwap:     53787
+	//   setReady:    34452
+	//   refund:      46692
+	//   relayClaim: 130507
+	//
+	const (
+		aliceGasRation = 150000 // roughly 10% more than newSwap+setRead+refund
+	)
+	// We give Alice enough gas money to refund if needed, but not enough to
+	// relay a claim:
+	//    150000 - (53787 + 34452) = 61761
+	//
```

Comment:
This didn't quite work on my first attempt. I had to add a minimum balance check to the relaying code, on top of this setup, to force the relaying over to a DHT relayer.  Otherwise the minimally funded Alice would try to relay using a low-ball gas price, and that would succeed in Ganache, but probably not on mainnet.